src/test/java/kitchenpos/application/MenuServiceTest.java

```diff
@@ -0,0 +1,216 @@
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(MockitoExtension.class)
+class MenuServiceTest {
+
+    MenuRepository menuRepository = new InMemoryMenuRepository();
+
+    MenuGroupRepository menuGroupRepository = new InMemoryMenuGroupRepository();
+
+
+    ProductRepository productRepository = new InMemoryProductRepository();
+
+    PurgomalumClient purgomalumClient = new FakePurgomalumClient();
+
+    MenuService menuService;
+
+    @BeforeEach
+    void setUp() {
+        menuService = new MenuService(menuRepository, menuGroupRepository, productRepository, purgomalumClient);
+    }
+
+    @Test
+    void create() {
+
+        //given
+        MenuGroup menuGroup = MenuGroupTestFixture.createMenuGroup(UUID.randomUUID(), "치킨류");
+        menuGroupRepository.save(menuGroup);
+        Product product = ProductTestFixture.createProduct(UUID.randomUUID(), "후라이드치킨", 18000L);
+        productRepository.save(product);
+
+        Menu request = MenuTestFixture.createMenuRequest("후라이드치킨", 18000L, true, product);
+        request.setMenuGroup(menuGroup);
+        request.setMenuGroupId(menuGroup.getId());
+
+        MenuProduct menuProduct = MenuProductTestFixture.createMenuProductRequest(1L, 1L, product);
+        request.setMenuProducts(List.of(menuProduct));
+        //when
+        Menu response = menuService.create(request);
+
+        //then
+        assertEquals(request.getName(), response.getName());
+        assertEquals(request.isDisplayed(), response.isDisplayed());
+
+
+    }
+
+    @Test
+    void mustHaveMenuGroup() {
+        //given
+        MenuGroup menuGroup = MenuGroupTestFixture.createMenuGroup(UUID.randomUUID(), "치킨류");
+        menuGroupRepository.save(menuGroup);
+        Product product = ProductTestFixture.createProduct(UUID.randomUUID(), "후라이드치킨", 18000L);
+        productRepository.save(product);
+
+        Menu request = MenuTestFixture.createMenuRequest("후라이드치킨", 18000L, true, product);
+
+        MenuProduct menuProduct = MenuProductTestFixture.createMenuProductRequest(1L, 1L, product);
+        request.setMenuProducts(List.of(menuProduct));
+
+        //when then
+        assertThrows(NoSuchElementException.class, () -> menuService.create(request));
+
+    }
+
+    @Test
+    void mustHaveProduct() {
+        //given
+        MenuGroup menuGroup = MenuGroupTestFixture.createMenuGroup(UUID.randomUUID(), "치킨류");
+        menuGroupRepository.save(menuGroup);
+        Product product = ProductTestFixture.createProduct(UUID.randomUUID(), "후라이드치킨", 18000L);
+        productRepository.save(product);
+
+        Menu request = MenuTestFixture.createMenuRequest("후라이드치킨", 18000L, true, new Product());
+        request.setMenuGroup(menuGroup);
+        request.setMenuGroupId(menuGroup.getId());
+
+        //when then
+        assertThrows(IllegalArgumentException.class, () -> menuService.create(request));
```

Comment:
assertThrows도 나쁘지 않지만 저는 개인적으로 assertJ의 `assertThatThrownBy`가 조금 더 직관적이여서 이 검증문을 조금 더 선호하는 편이에요. 🙂

```suggestion
        assertThatThrownBy(() -> menuService.create(request))
            .isExactlyInstanceOf(IllegalArgumentException.class);
```